### PR TITLE
Making asset unavailable for purchase if the size is zero

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Closes:

Changes proposed in this PR:
- Checking if the file size is 0
- Making the asset unavailable if the file size is zero

### CODE REVIEW

File: src/components/organisms/AssetActions/Consume.tsx [hunk 1]
```diff
@@ -167,7 +167,11 @@ export default function Consume({
         </div>
         <div className={styles.pricewrapper}>
           <Price price={price} conversion />
-          {!isInPurgatory && <PurchaseButton />}
+          {parseInt(file?.contentLength) === 0 ? (
```
Comment (kremalicious) [inline src/components/organisms/AssetActions/Consume.tsx:170]: `file` here refers to the file metadata object within the DDO, which is only set upon publish, retrieved when file is added. So this check here would only kick in if file has been published with `contentLength: 0` which is very unlikely

**Comment by kremalicious** [comment]
Feels like this is generally a better direction for solving #452, which could make #463 obsolete. If we can figure out file availability automatically by hitting the respective `provider` endpoint (instead of basing logic on fixed `file` metadata) then we can remove button as is done in this PR. If we have this there would not be much reasons to have publisher pay gas fees to manually disable/enable the button.

File: src/components/organisms/AssetActions/Consume.tsx (hunk 1 above)
Comment (bogdanfazakas) [reply src/components/organisms/AssetActions/Consume.tsx:170]: I think we can use this provide POST method (https://docs.oceanprotocol.com/references/provider/#apiv1servicesfileinfo) to get the file Content-Type and Content-Length using the assed did

**Comment by kremalicious** [comment]
As outlined in #479 we can't rely on `content-length` being correct all the time as it looks like. Considering this, informing user instead of disabling action in #479 seems like a better direction so closing this

**Pull request closed without merging**